plugin/src/main/java/org/opensearch/ml/action/update/UpdateModelTransportAction.java

```diff
@@ -0,0 +1,564 @@
+
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_GROUP_INDEX;
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_INDEX;
+import static org.opensearch.ml.common.FunctionName.REMOTE;
+import static org.opensearch.ml.common.FunctionName.TEXT_EMBEDDING;
+import static org.opensearch.ml.settings.MLCommonsSettings.ML_COMMONS_TRUSTED_CONNECTOR_ENDPOINTS_REGEX;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.opensearch.OpenSearchStatusException;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.DocWriteResponse;
+import org.opensearch.action.get.GetResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.action.update.UpdateResponse;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.MLModelGroup;
+import org.opensearch.ml.common.connector.Connector;
+import org.opensearch.ml.common.model.MLModelState;
+import org.opensearch.ml.common.transport.update.MLUpdateModelAction;
+import org.opensearch.ml.common.transport.update.MLUpdateModelCacheAction;
+import org.opensearch.ml.common.transport.update.MLUpdateModelCacheNodesRequest;
+import org.opensearch.ml.common.transport.update.MLUpdateModelInput;
+import org.opensearch.ml.common.transport.update.MLUpdateModelRequest;
+import org.opensearch.ml.engine.MLEngine;
+import org.opensearch.ml.helper.ConnectorAccessControlHelper;
+import org.opensearch.ml.helper.ModelAccessControlHelper;
+import org.opensearch.ml.model.MLModelGroupManager;
+import org.opensearch.ml.model.MLModelManager;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class UpdateModelTransportAction extends HandledTransportAction<ActionRequest, UpdateResponse> {
```

Comment:
General question: 
For now, the update model API will always update model index and model cache. Should we allow user to control only update model index ?

Reply:
We can raise the concern in a separate issue and lead the discussion to the community. We can use setting flags or separate api to implement this.